libbpf-tools/compat.h

```diff
@@ -21,4 +23,24 @@ int bpf_buffer__open(struct bpf_buffer *buffer, bpf_buffer_sample_fn sample_cb,
 int bpf_buffer__poll(struct bpf_buffer *, int timeout_ms);
 void bpf_buffer__free(struct bpf_buffer *);
 
+/* taken from libbpf */
+#ifndef __has_builtin
+#define __has_builtin(x) 0
+#endif
+
+static inline void *libbpf_reallocarray(void *ptr, size_t nmemb, size_t size)
+{
+	size_t total;
+
+#if __has_builtin(__builtin_mul_overflow)
+	if (unlikely(__builtin_mul_overflow(nmemb, size, &total)))
```

Comment:
could you remove 'unlikely' here? The 'unlikely' is defined in libbpf_internal.h and not available to users using libbpf standard uapi functions.

This can be easily reproduced with 'make CC=clang' where the clang points to latest clang which does have __builtin_mul_overflow builtin.

```
In file included from compat.c:4:
./compat.h:36:6: warning: call to undeclared function 'unlikely'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
        if (unlikely(__builtin_mul_overflow(nmemb, size, &total)))
            ^
1 warning generated.
...
/usr/bin/ld: Dwarf Error: Invalid or unhandled FORM value: 0x25.
/home/yhs/work/bcc/libbpf-tools/.output/biotop.o: In function `grow_vector':
biotop.c:(.text+0x41): undefined reference to `unlikely'
/home/yhs/work/bcc/libbpf-tools/.output/biotop.o: In function `main':
```

Reply:
Done.